src/main/java/com/iridium/iridiumskyblock/listeners/PlayerInteractListener.java

```diff
@@ -39,15 +32,33 @@ public void onClick(PlayerInteractEvent event) {
             return;
         }
 
-        event.getClickedBlock().setType(Material.AIR);
-        if (itemInHand.getAmount() > 1) {
-            itemInHand.setAmount(itemInHand.getAmount() - 1);
-            player.getInventory().addItem(new ItemStack(Material.LAVA_BUCKET)).values().forEach(itemStack ->
-                    player.getWorld().dropItem(player.getLocation(), itemStack)
-            );
-        } else {
-            itemInHand.setType(Material.LAVA_BUCKET);
+        if(IridiumSkyblock.getInstance().getConfiguration().obsidianBucket
+                && event.getAction().equals(Action.RIGHT_CLICK_BLOCK)
+                && event.getClickedBlock().getType().equals(Material.OBSIDIAN)
+                && itemInHand.getType().equals(Material.BUCKET)) {
+
+            event.getClickedBlock().setType(Material.AIR);
+
+            if (itemInHand.getAmount() > 1) {
+                itemInHand.setAmount(itemInHand.getAmount() - 1);
+                player.getInventory().addItem(new ItemStack(Material.LAVA_BUCKET)).values().forEach(itemStack ->
+                        player.getWorld().dropItem(player.getLocation(), itemStack)
+                );
+            } else {
+                itemInHand.setType(Material.LAVA_BUCKET);
+            }
+        }
+
+        if(IridiumSkyblock.getInstance().getConfiguration().endPortalPick
+                && (event.getAction().equals(Action.LEFT_CLICK_BLOCK) && player.isSneaking())
+                && event.getClickedBlock().getType().equals(Material.END_PORTAL_FRAME)
+                && (itemInHand.getType().name().contains("PICKAXE"))) {
+
+            event.getClickedBlock().getDrops().add(new ItemStack(Material.END_PORTAL_FRAME));
+            event.getClickedBlock().breakNaturally();
+
+            player.getInventory().addItem(new ItemStack(Material.END_PORTAL_FRAME)).values().forEach(itemStack ->
+                    player.getWorld().dropItem(player.getLocation(), itemStack));
```

Comment:
Why are we adding this to the drops and the player's inventory?

Reply:
adding to the drops doesn't do anything, i'll remove that line